fontir/src/variations.rs

```diff
@@ -364,74 +401,93 @@ struct LocationSortKey {
+                    .unwrap_or_default();
+                let lower = tent.lower.into_inner();
+                let peak = tent.peak.into_inner();
+                let upper = tent.upper.into_inner();
+
+                // If we're at the peak by definition we have full influence
+                if v == peak {
+                    return scalar; // *= 1
+                }
 
-                    // If the Tent is 0,0,0 it's always in full effect
-                    if (lower, peak, upper) == (ZERO, ZERO, ZERO) {
-                        return scalar; // *= 1
-                    }
+                // If the Tent is 0,0,0 it's always in full effect
+                if (lower, peak, upper) == (ZERO, ZERO, ZERO) {
+                    return scalar; // *= 1
+                }
 
-                    if v <= lower || upper <= v {
-                        trace!(
-                            "  {:?} => 0 due to {} {:?} at {:?}",
-                            self,
-                            axis_name,
-                            tent,
-                            location
-                        );
-                        return ZERO;
-                    }
+                if v <= lower || upper <= v {
+                    trace!(
+                        "  {:?} => 0 due to {} {:?} at {:?}",
+                        self,
+                        axis_name,
+                        tent,
+                        location
+                    );
+                    return ZERO;
+                }
 
-                    let subtract_me = if v < peak {
-                        tent.lower.into_inner()
-                    } else {
-                        tent.upper.into_inner()
-                    };
-                    scalar * (v - subtract_me) / (peak - subtract_me)
-                });
+                let subtract_me = if v < peak {
+                    tent.lower.into_inner()
+                } else {
+                    tent.upper.into_inner()
+                };
+                scalar * (v - subtract_me) / (peak - subtract_me)
+            },
+        );
         scalar
     }
+
+    pub fn insert(&mut self, axis_name: String, tent: Tent) {
+        if tent.has_non_zero() {
+            self.active_axes.insert(axis_name.clone());
+        }
+        self.axis_tents.insert(axis_name, tent);
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = (&String, &Tent)> {
+        self.axis_tents.iter()
+    }
+
+    pub fn is_default(&self) -> bool {
+        self.active_axes.is_empty()
+    }
 }
 
 /// The min/peak/max of a masters influence.
 ///
 /// Visualize as a tent of influence, starting at lower, peaking at peak,
 /// and dropping off to zero at upper.
-#[derive(Serialize, Deserialize, Clone, PartialEq, Eq)]
-struct Tent {
-    lower: NormalizedCoord,
-    peak: NormalizedCoord,
-    upper: NormalizedCoord,
+#[derive(Serialize, Deserialize, Default, Clone, PartialEq, Eq)]
+pub struct Tent {
+    pub lower: NormalizedCoord,
```

Comment:
is this terminology borrowed from somewhere else? In the spec I see these called min/max/peak iirc.

Reply:
Terminology was from the FontTools code itw as ported from. I agree min/peak/max is more common, will update.

Reply:
In the spec, I actually see these named "start" and "end", not "min" and "max", see the discussion about intermediate regions in https://learn.microsoft.com/en-us/typography/opentype/spec/otvaroverview

arguably "max" could be a bit ambiguous when placed next to "peak".